src/HotChocolate/Neo4J/src/Data/Language/Visitor/CypherVisitor.cs

```diff
@@ -0,0 +1,122 @@
+using System.Collections.Generic;
+using HotChocolate.Language;
+
+namespace HotChocolate.Data.Neo4J.Language
+{
+    public partial class CypherVisitor
+    {
+        private readonly LinkedList<IVisitable> _currentVisitedElements = new();
+
+        /// <summary>
+        /// Target for all rendered parts of the cypher statement.
+        /// </summary>
+        private readonly CypherWriter _writer = new();
```

Comment:
This is a memory leak. The writer implements IDispoable and gets StringBuilders from the pool. But the cypher visitor is not Disposable and the builder will never be returned back to the pool